### PR TITLE
feat(duckduckgo): init

### PR DESCRIPTION
**Pull request by GenShibe**
## 🎉 Theme for [DuckDuckGoose](https://duckduckgo.com) 🎉
it's a search engine that goes *QUACK*.

## 💬 Additional Comments 💬
use the default light mode for best experiences ty.

## 🗒 Checklist 🗒

- [x] I have read and followed Catppuccin's [submission guidelines](https://github.com/catppuccin/userstyles/blob/main/docs/userstyle-creation.md).
- [x] I have made a new directory underneath `/styles/<name-of-website>` containing the contents of the [`/template`](https://github.com/catppuccin/userstyles/blob/main/template/) directory.
  - [x] I have ensured that the new directory is in **lower-kebab-case**.
  - [x] I have followed the template and kept the preprocessor as [LESS](https://lesscss.org/#overview).
- [x] I have made sure to update the
      [`userstyles.yml`](https://github.com/catppuccin/userstyles/blob/main/src/userstyles.yml)
      file with information about the new userstyle.
- [x] I have included the following files:
  - `catppuccin.user.css` - all the CSS for the userstyle, based on the
    template.
  - `assets/mocha.webp`, `assets/macchiato.webp`, `assets/frappe.webp` and
    `assets/latte.webp` - individual screenshots of the themed website, in all 4
    Catppuccin flavors.
  - `catwalk.webp` - image of all individual screenshots stitched together,
    generated via [catwalk](https://github.com/catppuccin/toolbox#-catwalk).

### CODE REVIEW

**Review by sgoudham — COMMENTED** [review]
Just leaving this here, we should hold off on merging this until I/we figure out how to display this style in the README of `catppuccin/catppuccin`

Essentially, it'll end up looking a bit dumb right now if 2 `DuckDuckGo`'s appear on the README so we might need to modify how its represented over there or maybe not merge it here (which I just learned is similar to `catppuccin/youtubemusic`)

**Comment by andreasgrafen** [comment]
The bold statement would be that both DDG and youtubemusic are just packaged website that imitate native apps. Hence: Move everything into the userstyles repo. But then it could be argued that something like VS Code also is just a website when it comes down to it.

Anyhow; a little out of scope, but for youtubemusic it's basically the same stylesheet, no?  At that point I'd actually argue that it may be better to host is as a userstyle with bonud instructions as to how you can inject it into the packaged thing. But most people I know that use youtube music are using it in the browser anyway.
The same argument could be made about DDG as well.

Alternatively it could just be tagged accordingly like
- DuckDuckGo (Web)
- DuckDuckGo (Native)

**Comment by sgoudham** [comment]
But the larger question is can we avoid having 2 "listings" of the same application :thinking:

**Comment by andreasgrafen** [comment]
> But the larger question is can we avoid having 2 "listings" of the same application 🤔

By just having them as userstyles first and foremost – and if there are possibilities to inject them into a packaged desktop app, just put them as extra instructions. And we could still indicate that with an emoji key or something for the items where that's the case.

Would it be smart to move this discussion elsewhere since it's not really about the PR? xd

File: styles/duckduckgo/catppuccin.user.css
Comment (uncenter) [inline styles/duckduckgo/catppuccin.user.css:19]:
```suggestion
@-moz-document domain('duckduckgo.com') {
```

File: styles/duckduckgo/catppuccin.user.css
Comment (uncenter) [inline styles/duckduckgo/catppuccin.user.css:663]:
```suggestion
```

File: styles/duckduckgo/catppuccin.user.css
Comment (uncenter) [inline styles/duckduckgo/catppuccin.user.css:691]:
```suggestion
    @media (prefers-color-scheme: light) {
        :root {
            #catppuccin(@lightFlavor, @accentColor);
        }
    }
    @media (prefers-color-scheme: dark) {
        :root {
            #catppuccin(@darkFlavor, @accentColor);
        }
    }
```

**Review by uncenter — CHANGES_REQUESTED** [review]

**Review by night-lake — DISMISSED** [review]
Alright so just a few issues I noticed:
- The search button on the main page and search result titles default to blue accent and do not respect the set accent
- This. Just this. No comment.
Why is it in a box?
- Filters aren't themed either
- Some unthemed text
	- image
	- image



(I am able to fix these, just making a list so it's known what needs to be looked at :P)

**Comment by GenShibe** [comment]
> Alright so just a few issues I noticed:
> 
> * The search button on the main page and search result titles default to blue accent and do not respect the set accent
> * This. Just this. No comment.
>   Why is it in a box?
> * Filters aren't themed either
> * Some unthemed text
>   
>   * image
>   * image
> 
> (I am able to fix these, just making a list so it's known what needs to be looked at :P)

if you want to fix them, go right ahead, not going to stop you lol, i'll get to them probably tmr or friday

**Comment by GenShibe** [comment]
> * The search button on the main page and search result titles default to blue accent and do not respect the set accent


regarding the links, i discussed this with the userstyle queen herself, bell, and she said that following google and having blue links was ok so i'm leaving that part as is

**Comment by nekowinston** [comment]
> Just leaving this here, we should hold off on merging this until I/we figure out how to display this style in the README of `catppuccin/catppuccin`
> 
> Essentially, it'll end up looking a bit dumb right now if 2 `DuckDuckGo`'s appear on the README so we might need to modify how its represented over there or maybe not merge it here (which I just learned is similar to `catppuccin/youtubemusic`)

Just saw this comment, I thought the intention was to archive DuckDuckGo, and I commented as much on an issue over there yesterday. I'd be fine with doing that and taking ctp/duckduckgo off the port index.

**Comment by night-lake** [comment]
> > * The search button on the main page and search result titles default to blue accent and do not respect the set accent
> 
> regarding the links, i discussed this with the userstyle queen herself, bell, and she said that following google and having blue links was ok so i'm leaving that part as is

I feel like it should be an option on whether to use blue or the accent, unless that is bell territory.

**Comment by GenShibe** [comment]
> > > * The search button on the main page and search result titles default to blue accent and do not respect the set accent
> 
> > 
> 
> > regarding the links, i discussed this with the userstyle queen herself, bell, and she said that following google and having blue links was ok so i'm leaving that part as is
> 
> 
> 
> I feel like it should be an option on whether to use blue or the accent, unless that is bell territory.

@isabelroses ahem, belllllll

**Comment by isabelroses** [comment]
> I feel like it should be an option on whether to use blue or the accent, unless that is bell territory.

Following catppuccin's color guidelines we should be using blue for links.

**Comment by sgoudham** [comment]
Is this ready to be merged in btw? @GenShibe @isabelroses 

If it is, then we can merge it in and archive the existing catppuccin/duckduckgo repository as winston previously mentioned.

**Comment by isabelroses** [comment]
> Is this ready to be merged in btw? @GenShibe @isabelroses
> 
> If it is, then we can merge it in and archive the existing catppuccin/duckduckgo repository as winston previously mentioned.

I think its good to merge unless gen wants to work on anything else.

**Comment by GenShibe** [comment]
> > Is this ready to be merged in btw? @GenShibe @isabelroses
> > If it is, then we can merge it in and archive the existing catppuccin/duckduckgo repository as winston previously mentioned.
> 
> I think its good to merge unless gen wants to work on anything else.

pretty sure i've tackled the major stuff so it should be fine

**Review by isabelroses — APPROVED** [review]

**Pull request merged**
